examples/find_test_2022_t0_times.py

```diff
@@ -0,0 +1,63 @@
+"""This script finds the 2022 test set t0 times and saves them to the cloudcasting package."""
+
+import importlib.util
+import os
+
+import numpy as np
+import pandas as pd
+import xarray as xr
+
+from cloudcasting.dataset import find_valid_t0_times
+from cloudcasting.download import _get_sat_public_dataset_path
+from cloudcasting.validation import DATA_INTERVAL_SPACING_MINUTES, FORECAST_HORIZON_MINUTES
+
+# Set a max history length which we will support in the validation process
+# We will not be able to fairly score models which require a longer history than this
+# But by setting this too long, we will reduce the samples we have to score on
+
+# The current FORECAST_HORIZON_MINUTES is 3 hours so we'll set this conservatively to 6 hours
+MAX_HISTORY_MINUTES = 6 * 60
+
+# Open the 2022 dataset
+ds = xr.open_zarr(_get_sat_public_dataset_path(2022, is_hrv=False))
+
+# Filter to defined time frequency
+mask = np.mod(ds.time.dt.minute, DATA_INTERVAL_SPACING_MINUTES) == 0
+ds = ds.sel(time=mask)
+
+# Mask to the odd fortnights - i.e. the 2022 test set
+mask = np.mod(ds.time.dt.dayofyear // 14, 2) == 1
+ds = ds.sel(time=mask)
+
+
+# Find the valid t0 times
+valid_t0_times = find_valid_t0_times(
+    datetimes=pd.DatetimeIndex(ds.time),
+    history_mins=MAX_HISTORY_MINUTES,
+    forecast_mins=FORECAST_HORIZON_MINUTES,
+    sample_freq_mins=DATA_INTERVAL_SPACING_MINUTES,
+)
+
+# Print the valid t0 times to sanity check
+print(f"Number of available t0 times: {len(valid_t0_times)}")
+print(f"Actual available t0 times: {valid_t0_times}")
+
+
+# Find the path of the cloudcasting package so we can save the valid times into it
+spec = importlib.util.find_spec("cloudcasting")
+if spec and spec.origin:
+    package_path = os.path.dirname(spec.origin)
+else:
+    msg = "Path of package `cloudcasting` can not be found"
+    raise ModuleNotFoundError(msg)
```

Comment:
This seems a bit messy. Is cloudcasting expecting to load these files from a directory inside the package? How about using standardised data directories (e.g. from [appdirs](ansible_vars_file))?

Reply:
It is expecting to load the files from within the package. We need to make sure everyone is scoring the their models on the same set of datetimes so we include the list of datetimes in a csv file. 

I don't seem to have access to appdirs but I'm open to any cleaner way of doing this

Reply:
Sorry - link was broken (fixed now). I suggest something like `appdirs.user_data_dir(appname, appauthor)` might make sense here (will use a path like:  `/Users/trentm/Library/Application Support/appname` on macOS or similarly appropriate path on other OSes).

Reply:
Maybe I'm misunderstanding something here - aren't we trying to save a list of times to a location that the cloudcasting package can read later?

If so, I'm just suggesting that we save it to `appdirs.user_data_dir("cloudcasting", "cloudcasting")` instead of to `/path/to/cloudcasting/module/data`. If the `cloudcasting` module reads it from `appdirs.user_data_dir("cloudcasting", "cloudcasting")` too then that should give the same path.

Reply:
Ahhh, I think its my misunderstanding and I realise I haven't explained this well. 

We include the outputs from this script in the library already. It is saved under `cloudcasting/data/test_2022_t0_times.csv.zip`. We load [this file in the `ValidationSatelliteDataset`](https://github.com/alan-turing-institute/cloudcasting/blob/facc39b93b3ba441ea3f1318c6f6555e42d2e0ce/src/cloudcasting/dataset.py#L272-L274) to specify which times should be used in the validation process - so all models run on the exact same set of times. 

The user should never be running this script themselves, so saving it to user_data_dir doesn't seem quite right. The script is included so that we can trace how the times in the `cloudcasting/data/test_2022_t0_times.csv.zip` file were generated. These times should never be replaced from this point forward but I think its still good to archive this script and be able to trace how these times were generated. 

Maybe there is a better approach here. Maybe in the script we should just save it into the current working directory? As it stands, if the user ran this script it would replace their t0 times and potentially corrupt their version of the package. What do you think?

Reply:
The idea of `appdirs` is to give you a consistent per-application directory for settings, that only depends on the name of the application. I think this would still work here (e.g. reading and writing from a single directory) - up to you whether (a) you think this is worth doing and (b) whether it's a user/site data/config/whatever directory.

Reply:
That sounds good, but I think I'm still just unclear about how we get this list of times to the user. Would we need the user to run this script themselves or for this script to be run whilst installing the library or something?

Reply:
I think I see why this needs to *not* be run by a user and why `appdirs` is a bad fit. Perhaps this could be e.g. a GitHub Action that runs in CI and generates a PR if these times need to be updated?